dd-sdk-android/src/main/kotlin/com/datadog/android/telemetry/internal/TelemetryEventHandler.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.telemetry.internal
+
+import com.datadog.android.core.internal.persistence.DataWriter
+import com.datadog.android.core.internal.time.TimeProvider
+import com.datadog.android.core.internal.utils.loggableStackTrace
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.internal.domain.event.RumEventSourceProvider
+import com.datadog.android.rum.internal.domain.scope.RumRawEvent
+import com.datadog.android.telemetry.model.TelemetryDebugEvent
+import com.datadog.android.telemetry.model.TelemetryErrorEvent
+
+internal class TelemetryEventHandler(
+    private val serviceName: String,
+    private val sdkVersion: String,
+    private val sourceProvider: RumEventSourceProvider,
+    private val timeProvider: TimeProvider
+) {
+
+    fun handleEvent(event: RumRawEvent.SendTelemetry, writer: DataWriter<Any>) {
+
+        val timestamp = event.eventTime.timestamp + timeProvider.getServerOffsetMillis()
+
+        val context = GlobalRum.getRumContext()
+        val activeViewId = context.viewId.orEmpty()
+        val activeActionId = context.actionId.orEmpty()
+        val sessionId = context.sessionId
+        val applicationId = context.applicationId
+
+        val telemetryEvent: Any = when (event.type) {
+            TelemetryType.DEBUG -> {
+                TelemetryDebugEvent(
+                    dd = TelemetryDebugEvent.Dd(),
+                    date = timestamp,
+                    source = sourceProvider.telemetryDebugEventSource
+                        ?: TelemetryDebugEvent.Source.ANDROID,
+                    service = serviceName,
+                    version = sdkVersion,
+                    application = TelemetryDebugEvent.Application(applicationId),
+                    session = TelemetryDebugEvent.Session(sessionId),
+                    view = TelemetryDebugEvent.View(activeViewId),
+                    action = TelemetryDebugEvent.Action(activeActionId),
```

Comment:
view and actions are optional in the schema, so we should make them null when no viewid/actionid is available instead of sending an empty string